packages/kos-sdk/src/chains/tron/mod.rs

```diff
@@ -213,15 +273,17 @@ impl TRX {
 
         let tx: kos_proto::tron::Transaction = match options.token {
             Some(token) if token != "TRX" => {
-                // todo!() check if TRC20 transfer
-                let contract = kos_proto::tron::TransferAssetContract {
-                    owner_address: addr_sender.as_bytes().to_vec(),
-                    to_address: addr_receiver.as_bytes().to_vec(),
-                    amount: amount.to_i64(),
-                    asset_name: token.as_bytes().to_vec(),
-                };
-
-                requests::create_asset_transfer(&node, contract).await?
+                // Check if TRC20 transfer
+                let valid_address = TRX::validate_address(&token, None)?;
+                Self::check_valid_address(
+                    &valid_address,
+                    &addr_sender,
+                    &addr_receiver,
+                    &amount,
+                    &token,
+                    &node,
+                )
+                .await?
```

Comment:
should split paths here, one for trigger contract and another to TRC10 transfers... not one function for both... also, try keep function names to the action it peforms